# Hypothesis Testing 
## Pattern
##### 1. Statement
##### 2. State the Hypothesis (Null hypothesis & Alternate hypothesis)
##### 3. Select a Significance Level
##### 4. Compute the Test Statistic (t-test in this problem statement). Basically it can be any metric (Univariate or Bivariate)
##### 5. Determine the p-value
##### 6. Make a Decision
##### 7. Interpret the Results

In [3]:
import pandas as pd 

In [4]:
dataset=pd.read_csv('PrePlacement.csv')

In [5]:
dataset

sl_no gender  ssc_p    ssc_b  hsc_p    hsc_b     hsc_s  degree_p  \
0        1      M  67.00   Others  91.00   Others  Commerce     58.00   
1        2      M  79.33  Central  78.33   Others   Science     77.48   
2        3      M  65.00  Central  68.00  Central      Arts     64.00   
3        4      M  56.00  Central  52.00  Central   Science     52.00   
4        5      M  85.80  Central  73.60  Central  Commerce     73.30   
..     ...    ...    ...      ...    ...      ...       ...       ...   
210    211      M  80.60   Others  82.00   Others  Commerce     77.60   
211    212      M  58.00   Others  60.00   Others   Science     72.00   
212    213      M  67.00   Others  67.00   Others  Commerce     73.00   
213    214      F  74.00   Others  66.00   Others  Commerce     58.00   
214    215      M  62.00  Central  58.00   Others   Science     53.00   

      degree_t workex  etest_p specialisation  mba_p      status    salary  
0     Sci&Tech     No     55.0         Mkt&HR  58.80      Placed  270000.0  
1     Sci&Tech    Yes     86.5        Mkt&Fin  66.28      Placed  200000.0  
2    Comm&Mgmt     No     75.0        Mkt&Fin  57.80      Placed  250000.0  
3     Sci&Tech     No     66.0         Mkt&HR  59.43  Not Placed       0.0  
4    Comm&Mgmt     No     96.8        Mkt&Fin  55.50      Placed  425000.0  
..         ...    ...      ...            ...    ...         ...       ...  
210  Comm&Mgmt     No     91.0        Mkt&Fin  74.49      Placed  400000.0  
211   Sci&Tech     No     74.0        Mkt&Fin  53.62      Placed  275000.0  
212  Comm&Mgmt    Yes     59.0        Mkt&Fin  69.72      Placed  295000.0  
213  Comm&Mgmt     No     70.0         Mkt&HR  60.23      Placed  204000.0  
214  Comm&Mgmt     No     89.0         Mkt&HR  60.22  Not Placed       0.0  

[215 rows x 15 columns]

### Statement

Question : Test whether ssc_p of male and hsc_p of male is same?

Here in this case - It is Paired T-Test / Dependent Sample / same group & different conditions

### State the Hypothesis

Null hypothesis = There is no similarity between 10th pass mark & 12th pass mark of male

Alternate hypothesis = There is similairty between 10th pass mark and 12th mark of male

### Select a Significance Level

If p < 0.05, Reject the null hypothesis automatically accept the alternate hypothesis --> There is  _____ 

If p > 0.05, Accept null hypothesis --> There is no ______

### Compute the Test Statistic 

In [6]:
from scipy.stats import ttest_rel
Male = dataset[dataset['gender']=='M']['ssc_p']
male1 = dataset[dataset['gender']=='M']['hsc_p']
ttest_rel(Male,male1)

TtestResult(statistic=0.6842940392688153, pvalue=0.4949370716000775, df=138)

### Determine the p-value

P-vlaue = 0.49
p > 0.05 

### Make a Decision

Accept the Null Hypothesis

### Interpret the Results

There is a no similarity between 10th pass mark of male and 12th pass mark of male

## Unpaired T-Test - Independent Sample 
### RULE : Different Groups & Same condition

##### Example : Diff Groups : gender = Male & Female , One Condition : salary

In [4]:
from scipy.stats import ttest_ind
male = dataset[dataset['gender'] == 'M']['salary']
female = dataset[dataset['gender'] == 'F']['salary']
ttest_ind(male,female)

TtestResult(statistic=2.0937842507001165, pvalue=0.03746230397067219, df=213.0)

Inference : 

T-Test = 2.09 , Both Male & Female have high similarity in salary range

P-Value = 0.03 = 3%, less than 0.05 = 5%, so we can consider the P-Value. we will reject the null hypothesis, accept the alternate hypothesis 

there is a similarity between MAle & Female based on salary

##### Example : Diff Groups : hsc_b = Others, Central , One Condition - salary

In [5]:
central = dataset[dataset['hsc_b'] == 'Central']['salary']
others = dataset[dataset['hsc_b'] == 'Others']['salary']
ttest_ind(central,others)

TtestResult(statistic=-0.30570032095155825, pvalue=0.7601313863865756, df=213.0)

Inference : 

A negative value for the T_Test indicates that the mean of the first group is slightly smaller than the mean of the second group, but the difference is very small (since the value is close to zero).

T-Test = -0.30, Both Central & Other boards are showing small differences/ high similarities when compared with salary

P-Value = 0.76 = 76%, greater than 5%, we will accept the null hypothesis, There is no similarity in salary based on education standards.

## Paired T-Test - Dependent Sample
### RULE : Same Group & Different conditions

##### Example : One Group : gender = Male , Diff Conditions : hsc_p, ssc_p

In [6]:
from scipy.stats import ttest_rel
male = dataset[dataset['gender'] == 'M']['hsc_p']
male1 = dataset[dataset['gender'] == 'M']['ssc_p']
ttest_ind(male,male1)

TtestResult(statistic=-0.4609004016724563, pvalue=0.6452330329980636, df=276.0)

Inference : 

A negative value for the T_Test indicates that the mean of the first group is slightly smaller than the mean of the second group, but the difference is very small (since the value is close to zero).

T-Test = -0.46, In both 10th & 12th percentages, Male scores are very similar with very low differences. 

P-Value = 0.64 = 64%, greater than 5%, so we will accept the null hypothesis, which shows there is no similarity between the 10th & 12th percentages for males.

##### Example : One Group :  hsc_s = commerce, Diff Conditions : mba_p, etest_p

In [7]:
from scipy.stats import ttest_rel
commerce = dataset[dataset['hsc_s'] == 'Commerce']['mba_p']
commerce1 = dataset[dataset['hsc_s'] == 'Commerce']['etest_p']
ttest_ind(commerce,commerce1)

TtestResult(statistic=-7.079422262550767, pvalue=1.844299454870846e-11, df=224.0)

Inference : 

A negative value for the T_Test indicates that the mean of the first group is slightly smaller than the mean of the second group, but the difference is 
very small 

T-Test = -7.07,  Percentages in mba & etest for the students who took commerce group in 12th std shows similar results in majority with fewer differences. 

P-Value = 1.84, greater than 5%, so we will accept the hypothesis, which shows there are no similarity between MBA & etest percentages for commerce students.